src/report/mermaid.js

```diff
@@ -83,40 +99,71 @@ const convertedSubgraphSources = (pCruiseResult) => {
   return lTree;
 };
 
-const focusHighlights = (pModules) => {
+/**
+ * @param {import("../../types/cruise-result").IModule[]} pModules
+ * @param {Map<string, string>} pMinifiedNames
+ */
+const focusHighlights = (pModules, pMinifiedNames) => {
   const lHighLightStyle = "fill:lime,color:black";
 
   return pModules
     .filter((pModule) => pModule.matchesFocus === true)
-    .reduce(
-      (pAll, pModule) =>
-        (pAll += `\nstyle ${hashNodeName(pModule.source)} ${lHighLightStyle}`),
-      ""
-    );
+    .reduce((pAll, pModule) => {
+      const lSource = pMinifiedNames.get(pModule.source) || pModule.source;
+      return (pAll += `\nstyle ${hashNodeName(lSource)} ${lHighLightStyle}`);
+    }, "");
 };
 
-const renderMermaidThing = (pCruiseResult) => {
-  const subgraphs = convertedSubgraphSources(pCruiseResult);
-  const edges = convertedEdgeSources(pCruiseResult);
+/**
+ * @param {import("../../types/cruise-result").IModule[]} pModules
+ * @param {Boolean} pMinify
+ * @return {Map<string, string>}
+ */
+const minifiedNames = (pModules, pMinify) => {
+  const names = new Map();
+  if (!pMinify) return names;
+
+  let lCount = 0;
+  pModules.forEach((pModule) => {
+    const lPaths = pModule.source.split("/");
+
+    for (let lIndex = 0; lIndex < lPaths.length; lIndex += 1) {
+      const lName = lPaths.slice(0, lIndex + 1).join("/");
+      if (!names.get(lName)) {
+        names.set(lName, lCount.toString());
```

Comment:
The hash of the node uses a simple numeric index. The reason is that if you increment it, there will be no duplication and the character limit will not make the value that large.